packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -96,6 +88,27 @@ export abstract class DeployRetrieveExecutor<
     }
   }
 
+  async setApiVersionOn(components: ComponentSet) {
+    // Check the SFDX configuration to see if there is an overridden api version.
+    // Project level local sfdx-config takes precedence over global sfdx-config at system level.
+    const userConfiguredApiVersion = await this.getUserConfiguredApiVersion();
+    
+    // If no user-configured Api Version is present, then get the version from the Org.
+    const orgApiVersion = await this.getOrgApiVersion();
+    components.apiVersion = userConfiguredApiVersion ?? orgApiVersion;
+  }
+
+  async getUserConfiguredApiVersion() {
+    return await ConfigUtil.getConfigValue('apiVersion') as
+    | string
+    | undefined;
```

Comment:
when you use `as` you are basically overriding what typescript knows about the type.  This might be better: 

    const apiVersion = await ConfigUtil.getConfigValue('apiVersion');
    return apiVersion ? String(apiVersion) : undefined;

This way the inferred type is the same (string | undefined) but we still lean on not overriding what typescript is telling us.

Reply:
Cool!  Updated.